networkx/classes/tests/dispatch_interface.py

```diff
@@ -108,71 +110,49 @@ def convert_from_nx(
         if preserve_graph_attrs:
             G.graph.update(graph.graph)
 
+        # add nodes
+        G.add_nodes_from(graph)
         if preserve_node_attrs:
-            G.add_nodes_from(graph.nodes(data=True))
+            for n, dd in G._node.items():
+                dd.update(graph.nodes[n])
         elif node_attrs:
-            G.add_nodes_from(
-                (
-                    node,
-                    {
-                        k: datadict.get(k, default)
-                        for k, default in node_attrs.items()
-                        if default is not None or k in datadict
-                    },
+            for n, dd in G._node.items():
+                dd.update(
+                    (k, graph._node[n].get(k, v))
+                    for k, v in node_attrs.items()
+                    if v is not None or k in graph._node[n]
                 )
-                for node, datadict in graph.nodes(data=True)
-            )
-        else:
-            G.add_nodes_from(graph)
-
-        if graph.is_multigraph():
-            if preserve_edge_attrs:
-                G.add_edges_from(
-                    (u, v, key, datadict)
-                    for u, nbrs in graph._adj.items()
-                    for v, keydict in nbrs.items()
-                    for key, datadict in keydict.items()
-                )
-            elif edge_attrs:
-                G.add_edges_from(
-                    (
-                        u,
-                        v,
-                        key,
-                        {
-                            k: datadict.get(k, default)
-                            for k, default in edge_attrs.items()
-                            if default is not None or k in datadict
-                        },
-                    )
-                    for u, nbrs in graph._adj.items()
-                    for v, keydict in nbrs.items()
-                    for key, datadict in keydict.items()
-                )
-            else:
-                G.add_edges_from(
-                    (u, v, key, {})
-                    for u, nbrs in graph._adj.items()
-                    for v, keydict in nbrs.items()
-                    for key, datadict in keydict.items()
-                )
-        elif preserve_edge_attrs:
-            G.add_edges_from(graph.edges(data=True))
+
+        # add edges
+        undirected = not graph.is_directed()
+        for n, nbrs in graph._adj.items():
+            Gadj_n = G._adj[n]
+            for nbr, keydict in nbrs.items():
+                if undirected and n in G._adj[nbr]:
```

Comment:
We are building the two sides of the edge -- one at a time. So this is the check that we've already loaded the other representative of the edge and we need to use that datadict instead of creating a new one. 

Said another way, the first time an edge hits this condition, it will be False because G hasn't added the edge yet. The second time it will be True.  This is the line that puts the edge info into the adjacency structure with the same order as the original adjacency. If you add both sides of the edge at the same time (e.g. by using `add_edges_from` the adjacency order can change).